manifests/init.pp

```diff
@@ -49,28 +55,45 @@
     default => $user
   }
 
+  $composer_group = $group ? {
+    'UNDEF' => $::composer::params::group,
+    default => $group
+  }
+
+  # common dependency
+  package { 'patch':
+    ensure => latest
+  }
+
+
+  # fetch composer into target_dir
   wget::fetch { 'composer-install':
     source      => $::composer::params::phar_location,
     destination => "${composer_target_dir}/${composer_command_name}",
-    execuser    => $composer_user,
+    execuser    => 'root',
   }
 
-  exec { 'composer-fix-permissions':
-    command => "chmod a+x ${composer_command_name}",
-    path    => '/usr/bin:/bin:/usr/sbin:/sbin',
-    cwd     => $composer_target_dir,
-    user    => $composer_user,
-    unless  => "test -x ${composer_target_dir}/${composer_command_name}",
+
+  # apply user and group permissions to downloaded composer using native puppet File type
```

Comment:
if you leave comments, then please don't tell us WHAT you're doing, but WHY you're doing this